projects/dotcms-js/src/lib/core/core-web.service.mock.ts

```diff
@@ -1,29 +1,103 @@
 import { Observable, of } from 'rxjs';
-import { RequestOptionsArgs, Http, Request } from '@angular/http';
-import { map } from 'rxjs/operators';
+import { map, filter } from 'rxjs/operators';
 import { ResponseView } from './util/response-view';
+import {
+    HttpClient,
+    HttpRequest,
+    HttpEvent,
+    HttpEventType,
+    HttpResponse,
+    HttpParams,
+    HttpHeaders
+} from '@angular/common/http';
+import { RequestMethod, RequestOptionsArgs } from '@angular/http';
+import { RequestOptionsParams } from 'dotcms-js';
 
 @Injectable()
 export class CoreWebServiceMock {
+    constructor(private _http: HttpClient) {}
 
-    constructor(private _http: Http) {}
+    request(options: RequestOptionsArgs): Observable<any> {
+        const optionsArgs: RequestOptionsParams = {
+            params: new HttpParams()
+        };
 
-    request(options: any): Observable<any> {
-        return this._http.request(new Request(options));
+        if (options.params) {
+            Object.keys(options.params).forEach((key) => {
+                optionsArgs.params = optionsArgs.params.set(key, options.params[key]);
+            });
+        }
+
+        return this._http
+            .request(
+                new HttpRequest(
+                    RequestMethod[options.method],
+                    options.url,
+                    options.body,
+                    { params: optionsArgs.params }
+                )
+            )
+            .pipe(
+                filter(<T>(event: HttpEvent<T>) => event.type === HttpEventType.Response),
+                map((resp: HttpResponse<any>) => {
+                    try {
+                        return resp.body;
+                    } catch (error) {
+                        return resp;
+                    }
+                })
+            );
     }
 
-    public requestView(options: RequestOptionsArgs): Observable<ResponseView> {
-        return this._http.request(new Request(<any>options)).pipe(
-            map((resp) => {
-                return new ResponseView(resp);
-            })
-        );
+    requestView(options: RequestOptionsArgs): Observable<ResponseView> {
+        const optionsArgs: RequestOptionsParams = {
+            headers: new HttpHeaders(),
+            params: new HttpParams()
+        };
+
+        if (options.params) {
+            Object.keys(options.params).forEach((key) => {
+                optionsArgs.params = optionsArgs.params.set(key, options.params[key]);
+            });
+        }
+
+        if (options.search) {
+            optionsArgs.params = this.setHttpParams(options.search, optionsArgs.params);
+        }
+
+        return this._http
+            .request(
+                new HttpRequest(RequestMethod[options.method], options.url, options.body, {
+                    params: optionsArgs.params
+                })
+            )
+            .pipe(
+                filter(<T>(event: HttpEvent<T>) => event.type === HttpEventType.Response),
```

Comment:
Not sure what this filter is for and also `<T>` where is coming from?

Reply:
https://github.com/dotCMS/core-web/pull/1429/files#r493914633

`<T>` is a generic used by the type `HttpEvent` which can be of multiple types, depending the BE response
https://angular.io/api/common/http/HttpEvent

Reply:
Exactly... but you need to add that generic in the function declaration:

```requestView<T>(options: RequestOptionsArgs)```

Because we have to pass when the functions is called.

https://www.typescriptlang.org/docs/handbook/generics.html

So like if we are using the `core-web.service` to get a contentlet we have to do something like:

```this.coreWebService.requestView<DotContentlet>({...}).subscribe(...)```

Like they do here: https://angular.io/guide/http#requesting-a-typed-response